crates/fluvio-protocol-derive/src/ast/prop.rs

```diff
@@ -16,6 +21,38 @@ pub(crate) struct UnnamedProp {
     pub attrs: PropAttrs,
 }
 
+pub fn validate_versions_tokens(
+    min_prop: Option<&PropAttrsType>,
+    max_props: Option<&PropAttrsType>,
+    field: Option<&str>,
+) -> TokenStream {
+    let min = prop_attrs_type_value(min_prop);
+    let max = prop_attrs_type_value(max_props);
+
+    match (max_props, field) {
```

Comment:
I don't understand the conditions here: we check `#min>#max` twice, we check `#min < 0` only if max_props is Some(_), why? We use `concat!` but don't do actual concatenation.